app/views/saved_queries/show.html.erb

```diff
@@ -0,0 +1,45 @@
+<div class="modal-body">
+  <h1>Saved Query</h1>
+  <hr/>
+  <div class="container-fluid">
+    <h3><strong><%= @saved_query.title %></strong></h3>
+  </div>
+  <div class="container-fluid">
+    <h5><%= @saved_query.description %></h5>
+  </div>
+  <div class="container-fluid">
+    <h5><%= @saved_query.public ? 'Public' : 'Not Public' %></h5>
+  </div>
+  <div class="pb-3"></div>
+  <%= form_tag("/query", :method => "get") do %>
+    <%= label_tag(:query, 'SQL Query:') %>
+    <% if @error.present? %>
+      <div id="error_explanation">
+        <ul>
+        <% @error.each do |message| %>
+          <li class="text-danger"><%= message %></li>
+        <% end %>
+        </ul>
+      </div>
+    <% end %>
+    <%= text_area_tag(:query, @saved_query.sql, :required => "required", :size =>"25x5") %>
+    <div class="pb-3"></div>
+    <div class="d-flex">
+      <div class="me-auto">
+        <button type="submit" class="btn btn-primary btn-sm">Execute SQL Query</button>
+      </div>
+      <div class="d-flex gap-3">
+        <div>
+          <%= link_to 'Edit', edit_saved_query_path(@saved_query), :class => "btn btn-primary btn-sm" %>
+        </div>
+        <div>
+          <%= link_to 'Delete', saved_query_path(@saved_query), :class => "btn btn-danger btn-sm", :method => :delete, data: { confirm: 'Are you sure?' } %>
```

Comment:
Only show the buttons if the logged in user is the one that created the saved query